src/app-logic/browser.js

```diff
@@ -0,0 +1,186 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+/**
+ * This file is in charge of handling the message managing between profiler.firefox.com
+ * and the browser internals. This is done through a WebChannel mechanism. This mechanism
+ * allows us to safely send messages between the browser and an allowed domain.
+ */
+
+/**
+ * The messages are typed as an object so that the "type" field can be extracted
+ * using the $Keys utility type.
+ */
+type MessageToBrowserObject = {|
+  STATUS_QUERY: {| type: 'STATUS_QUERY' |},
+  ENABLE_MENU_BUTTON: {| type: 'ENABLE_MENU_BUTTON' |},
+|};
+
+/**
+ * The messages are typed as an object so that the "type" field can be extracted
+ * using the $Keys utility type.
+ */
+type MessageFromBrowserObject = {|
+  STATUS_RESPONSE: {|
+    type: 'STATUS_RESPONSE',
+    menuButtonIsEnabled: boolean,
+  |},
+  ENABLE_MENU_BUTTON_DONE: {| type: 'ENABLE_MENU_BUTTON_DONE' |},
+|};
+
+// Extract out the different values. Exported for tests.
+export type MessageToBrowser = $Values<MessageToBrowserObject>;
```

Comment:
I don't actually extract the Keys for the `MessageToBrowser`, but kept the Object-style definition in order to be consistent on the definitions.